lib/msprime.c

```diff
@@ -2138,26 +2138,39 @@ compute_falling_factorial_log(double m)
 static double
 msp_compute_lambda_Xi_dirac(msp_t *self, unsigned int num_ancestors)
 {
-    unsigned int l;
+    unsigned int l, m;
     double psi = self->model.params.dirac_coalescent.psi;
     double c = self->model.params.dirac_coalescent.c;
     double ret = 0;
     double b = num_ancestors;
+    m = GSL_MIN(num_ancestors, 4);
```

Comment:
This is mixing code and declarations. The style here is to have a block of declarations first, then code. This keeps compatibility with older C compilers (but is mainly just keeping uniform style).